src/api/app/models/bs_request.rb

```diff
@@ -107,6 +108,21 @@ def check_creator
     errors.add(:creator, "Login #{user.login} is not an active user")
   end
 
+  def double_actions
+    bs_request_actions.each do |actionA|
+      bs_request_actions.each do |actionB|
+        next if actionA == actionB
+        next if actionA.type == "add_role" # the exception that proves the rule
+        next if actionA.type == "maintenance_incident" # we could extend the check for package link target
+        if actionA.type == actionB.type \
+           && actionA.target_project == actionB.target_project \
+           && actionA.target_package == actionB.target_package
+          errors.add(:invalid, "double action for #{actionB.try(:target_project)}/#{actionB.try(:target_package)} ")
+        end
+      end
+    end
```

Comment:
Besides, this probably can be done with a simple [rails validation](http://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_uniqueness_of)